### DISCUSSION TITLE
Can getServerSideProps replace getInitialProps?

### DISCUSSION THREAD

**Discussion by stegano** [vercel/next.js#17362] [Help] [answered]
I had a problem moving the code used in getInitialProps to getServerSideProps.

I updated the UserId to Redux State, and when moving to the next page, the UserId value in the Redux State should be referenced in gerServerSideProps to get UserProfile detail information. 
but, the Redux State can't unknown to the server. 

Is there a solution?

**Comment by lfades** [comment, marked as answer]
`getServerSideProps` only runs in Node.js, so if you have global state inside it you will need to return the state as props and then hydrate the store in the browser. Check out the [`with-redux` example](https://github.com/vercel/next.js/tree/canary/examples/with-redux) for more details on how to do that.

To answer the question: `getInitialProps` runs it both the browser and Node.js, it's included in the browser build whilst `getServerSideProps` isn't. It may be replaced very easily with `getServerSideProps` in most cases but there are exceptions when its execution relies on browser context.

**Reply by stegano** [reply]
Thanks for your answer. 😀
